services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/services/director_v2/_thin_client.py

```diff
@@ -95,11 +95,12 @@ async def _(
         ) -> Response:
             headers = {X_SIMCORE_USER_AGENT: simcore_user_agent}
 
-            return await self.client.delete(
+            response: Response = await self.client.delete(
```

Comment:
your problem here is that the client has no type. so doing this might fail if you use the wrong calls.
for later refactoring, I would recommend using a dataclass where you have typing, or define the client type in the base class.